MAINT: Remove numpydoc version upper bound

prakritim01 opened the pull request:
"This PR removes the <1.9.0 constraint for numpydoc in build_tools/update_environments_and_lock_files.py to allow CI to use newer versions. Note: I was unable to run the lockfile update script locally due to missing Conda environment tools. I am happy to let a maintainer (or the bot) update the lockfiles if needed." 
Fixed in numpy/numpydoc#622, which is included in the latest release: https://github.com/numpy/numpydoc/releases/tag/v1.10.0

prakritim01 committed:
MAINT: Remove numpydoc version upper bound

lucyleeow commented on build_tools/update_environments_and_lock_files.py:
```diff
@@ -84,15 +84,11 @@
     "pytest-cov": "<=6.3.0",
 }
 
-
```
Lint is probably failing due to this line.

lucyleeow reviewed (commented):
Thanks for the PR. Looks good except the lint failure.

I would recommend you read our contributing guide before making PRs: https://scikit-learn.org/dev/developers/contributing.html#id1

It would be useful for the reviewer if you included the following information:

Fixed in https://github.com/numpy/numpydoc/pull/622, which is included in the latest release: https://github.com/numpy/numpydoc/releases/tag/v1.10.0

lucyleeow reviewed (approved):
LGTM thanks.

virchan reviewed (approved):
LGTM. Thanks, @prakritim01.

@lucyleeow, could we merge this?

lucyleeow commented:
Just realised that we should probably update the conda lock files and make sure CI passes. I think the lock files are updated automatically regularly but probably better to do it in this PR.

I can do it later or you can @virchan if you wish.

virchan commented:
I'm not very familiar with the process for updating the conda lock files yet, so it would be great if you could handle that part, @lucyleeow. I'm very happy to review the changes and help make sure everything looks good once it's up.

prakritim01 committed:
Merge branch 'main' into update-numpydoc-ci

prakritim01 committed:
MAINT fix ruff formatting

prakritim01 committed:
:x
Merge branch 'update-numpydoc-ci' of https://github.com/prakritim01/scikit-learn into update-numpydoc-ci

lucyleeow committed:
update doc lock files

lesteve reviewed (changes requested):
We need to wait until https://github.com/scikit-learn/scikit-learn/pull/32902 is resolved.

At the moment for some unknown reason, the memory increases in the doc build and gets killed before it can complete, see https://github.com/scikit-learn/scikit-learn/pull/32902#issuecomment-3685435307 for more details.

Also general reminder when you update doc build lock-files you need to make sure to push a commit with `[doc build]` to run the full doc build 😉.

prakritim01 commented:
> We need to wait until #32902 is resolved.
> 
> At the moment for some unknown reason, the memory increases in the doc build and gets killed before it can complete, see [#32902 (comment)](https://github.com/scikit-learn/scikit-learn/pull/32902#issuecomment-3685435307) for more details.
> 
> Also general reminder when you update doc build lock-files you need to make sure to push a commit with `[doc build]` to run the full doc build 😉.

Thanks for the heads-up! I will wait until #32902 is resolved.

Once that is fixed, let me know if you need me to push an empty commit with [doc build] to trigger the full build, or if we should handle it differently then.

lesteve commented:
@prakritim01 you can push an empty commit with `[doc build]` in the commit message now. This would allow to double-check that the memory usage problem is still there.

prakritim01 committed:
MAINT trigger doc build [doc build]

prakritim01 committed:
:x
Merge branch 'update-numpydoc-ci' of https://github.com/prakritim01/scikit-learn into update-numpydoc-ci

lucyleeow commented:
@prakritim01 thanks, but after discussing with @lesteve, we generally prefer maintainers to update lock files.

lucyleeow committed:
revert conda lock files

lucyleeow commented:
Finding it too difficult to see whats changed in the lock files with the history. I've reverted the lock file changes, leaving only changes to `build_tools/update_environments_and_lock_files.py`. I will open a new PR to update the lock files.

lucyleeow commented:
Ah can't merge due to changes requested, @lesteve feel free to merge when you're happy. I will open a new PR for lock file changes.

prakritim01 commented:
> Yeah sorry @lesteve, I am realising this was not a good first PR, though I appreciate @prakritim01 has now taken some time to work on it.
> 
> @prakritim01 how do you feel about picking a different issue to work on?

Hi @lucyleeow, thank you for the update!

I completely understand—dependencies can be tricky. I appreciate you letting me know. I am happy to close this PR and pick up a different issue that might be a better fit.

Do you have any suggestions for another "good first issue" or documentation fix I could tackle?
